### DISCUSSION TITLE
Error from reading vtk files

### DISCUSSION THREAD

**Discussion by hugy888** [idaholab/moose#21453] [Q&A Meshing] [answered]
Hello,

The vtk file was made using dream3D and we wanted to read it with moose and assign each grain ID from vtk to a subdomain_id from moose.
But when I tried to load vtk file with following code:

`[Mesh]`
`type = FileMesh`
`file = new.vtk`
`dim = 3`
`elem_type = HEX8`
`[]`

Error shows up as followed:
![image](https://user-images.githubusercontent.com/60995897/176457668-d58d9419-10fb-46ee-8aa1-356bbb87cad2.png)

Would you please take a look? Thanks.

**Comment by GiudGiud** [comment, marked as answer]
Hello

Can you try a FileMeshGenerator instead of the old FileMesh syntax please
https://mooseframework.inl.gov/source/meshgenerators/FileMeshGenerator.html

if that doesnt work i recommend installing meshio and converting it to exodus

Guillaume
